app/uk/gov/hmrc/agentepayeregistration/validators/AgentEpayeRegistrationValidator.scala

```diff
@@ -73,8 +77,23 @@ object AgentEpayeRegistrationValidator {
       request.address.addressLine4.map(x => validCharsWithLimit(x, "address line 4", 35))
     ).flatten
 
+    validate(validators ++ optionalFieldValidators)
+  }
+
+  def validateDateRange(dateFrom: String, dateTo: String): Validated[Failure, Unit] = {
+    def dateInPast(param: String, paramName: String) =
+      isISODate(param)(paramName).andThen(_ => isInPast(parseISODate(param))(paramName))
```

Comment:
Do we have to parse the same date multiple times? Maybe we could use PathBinders to have endpoint parameters of `Date` type?

Reply:
The `dateFrom` and `dateTo` parameters now use a `QueryStringBinder` for `LocalDates`. This cleans up the parsing/validation a little bit, at the slight expense of a different JSON error response format when date parameters fail to parse (as it's generated by Play)